roles/workarounds-host/tasks/main.yml

```diff
@@ -0,0 +1,6 @@
+- name: Reload dnsmasq config, so that VM's DNS cache can be refreshed.
```

Comment:
Can we move this into the `tasks` section instead of a new role in this case? Thanks Anil

Example: https://pastebin.com/B8PERpg5

Reply:
Mostly I just think we don't need a role as we probably don't plan to re-use a workaround in other parts of the code and it seems at least specific to this playbook for now, so it can live there without a role.

Reply:
@dougbtv thanks for the review. The reason i created a seperate role is becase we might hit more issue that we need to workaround at host level and VM level, so we can keep adding those workarounds in their respective roles (workarounds-host, workarounds-vms (in future, if needed)). Ideally these bugs should be handle at their respective level, so i am hoping they will go away in future, in that case, it's easier to toss these roles as well. 
If you think it's overkill let me know, i can move it in the init.yml.

Reply:
Additional consideration: it might need `post_tasks` instead of tasks actually, depending on when you need to HUP dnsmasq. [post_tasks help article](https://www.oreilly.com/library/view/ansible-playbook-essentials/9781784398293/ch02s09.html)

Reply:
Anil -- in the case you think we'll need to reuse it -- I'm ok, but, I'd prefer to change the name of the role. Maybe something like "virthost-dnsmasq" or something like that

Reply:
@dougbtv i was thinking the same and started with that as the first place :), but the thought was that we can add any future workarounds also in the same role, so that's why i didn't name it specific to this workaround. 
Not sure what's the best practice here. If kube-ansible needs new workaround for some issue in future, we should create a new role or putting them in same role is considered okay as well? I am okay both ways.

Reply:
Mainly my feeling is -- if we're going to keep around to be reusable as a role, we should know specifically what it is. Workaround or not, the name "workaround" in the roles directory is too vague for me. The other role names give me an idea of what's there when I go to edit code. So, I guess I'm just not big on the nomenclature. 

...We also could just keep it as a task until we need to use it again, and then create a role at the point where it needs to go in multiple playbooks.

Reply:
In my experience roles should be well named, reusable chunks of logic similar to what you'd find in a function. You'd probably not create a "workaround" function.

If you want to use `workaround` I would suggest that should be added as a tag to the tasks being run, and then enable the `workaround` tag when necessary for a particular deployment/inventory item.

Reply:
@leifmadsen This workaround needs to be run everytime (except first time), so not really optional, so i didn't take the `tags` approach. I am no ansible expert, so let me put the requirement here, and if you can suggest the best practise that would be great -- I hit the issue and i added the workaround, but we might need more workaround in future at the host level as well as at the VM level. So i wanted to take a approach where these workarounds are easily maintainable (not scatter all over the place) and easy to remove once it's not required. They might not be resulable across the code base, so they might not fit for role, probably better for task? (but i don't really see much reusability of existing roles, most of the roles are specific task),

Reply:
I'd just put the task you want into the appropriate role that the task is working against. You actually do want to pepper these across the repo, because you want to apply tasks that make sense for that particular role.

You can use the tag mechanism to be able to find the changes later and pull them out if necessary. Even if the default is to always run them, then so be it.

I'm not a big fan of potpourri roles as they quickly become unwieldy without a clear direction of what they are really accomplishing. The scope then results in the role becoming a dumping ground.

If the task needs to be performed, then it's a necessary part of your workflow, and the task should show up in the role that it is most closely aligned to. Either use tags (or even just add a `# [WORKAROUND]` or better a ` # [TODO]` comment above the task to note why it exists, and when it can be removed (workaround no longer necessary).

Using a workaround role will cause issues in the future as well when you're not sure when to trigger the role. You'll have something that wants to happen during the initial setup, then something else in a post-deploy change, and another thing that happens at step 4. At that point, you just end up with like `role_predeploy_workarounds`, `role_step4_workarounds`, and `role_postdeploy_workarounds` which starts to look a bit silly.

Reply:
Also you say "everytime (except first time)" which is a perfect reason to use something dynamic like a tag. That way you can optionally `--skip-tags workaround` on the first deploy, and then run them on subsequent deploys.

Reply:
@leifmadsen make sense, thanks for sharing the thoughts. I think as of now, i will add this as a task with tag, that seems like a resonable approach.